pkg/framework/resource/filter.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+// Filter is an interface whose methods are used to arbitrarily filter resources and subresources in a Resources object. Filtering criteria can be anything at all, so long as the "accepted" resources return true, and the filtered out resources return false.
+type Filter interface {
+	Resource(*Resource) bool
+	SubResource(*SubResource) bool
+}
+
+// NewAndFilter that takes as argument a slice of filters, which then work as a logical AND. This filter will only return resources that pass through every filter in the argument slice. It returns a new Filter value.
+func NewAndFilter(filters ...Filter) Filter {
+	new := andFilter{}
+	new.Filters = append(new.Filters, filters...)
+	return &new
+}
+
+// NewOrFilter that takes as argument a slice of filters, which then work as a logical OR. This filter will return resources that pass through any one or more filters in the argument slice. It returns a new Filter value.
+func NewOrFilter(filters ...Filter) Filter {
+	new := orFilter{}
+	new.Filters = append(new.Filters, filters...)
+	return &new
+}
+
+type emptyFilter struct {
```

Comment:
This is definied privately, but not used in this file. I don't know if it's used anywhere else in the package, but I still find this a little dubious

EDIT: It's used nowhere.

Reply:
Should we delete it or make it public?